src/components/pages/StatedGraphForm.tsx

```diff
@@ -1,17 +1,16 @@
 import React from 'react';
+import Chart from '../Chart';
+import { observer } from 'mobx-react-lite';
+import BalanceService from '../BalanceService';
 
-import ChartFunctional from '../ChartFunctional';
-let GraphForm = () => {
+const GraphForm = (balance) => {
```

Comment:
use better short version:

export const GraphForm = ...

Reply:
also, as I understand balance should be named balances

Reply:
Also as you said, GraphForm should be here only observable (I think that App gets fully rerendered if it is the only observable).
I'm not sure how you want to eliminate props drilling without using globals - and I am not sure if elimination is needed.